spec/models/event_schedule_spec.rb

```diff
@@ -0,0 +1,44 @@
+require 'rails_helper'
+
+RSpec.describe EventSchedule, type: :model do
+  describe '.populate_upcoming_show_and_tell!' do
+    subject { described_class.populate_upcoming_show_and_tell! }
+    before { travel_to Date.new(2017, 1, 1) }
+    after { travel_back }
+
+    let(:fridays) do
+      counter = 0
+      dates = []
+      4.times do
+        dates << Date.new(2017, 1, 6 + counter)
+        counter += 7
+      end
+      dates
+    end
+
+    it 'creates show and tell events' do
+      expect { subject }
+        .to change { Event.where(name: 'Show & Tell').length }
+    end
+
+    it 'creates show and tell events that take place on Friday' do
+      subject
+      Event.where(name: 'Show & Tell').each do |event|
+        expect(event.date.strftime('%A')).to eq('Friday')
+      end
+    end
+
+    it 'creates an event for each Friday in the upcoming four weeks' do
+      subject
+      Event.where(name: 'Show & Tell').each do |event|
+        expect(fridays).to include(event.date)
+      end
+    end
+
+    it 'doesnt create duplicate events' do
+      subject
+      expect { subject }
+        .not_to change { Event.where(name: 'Show & Tell').length }
```

Comment:
It doesn't matter too much in this test but usually prefer `count` or even `size` to length since `length` will load the full collection into memory.

This post does a reasonable job of explanaing the differences:
http://mensfeld.pl/2014/09/activerecord-count-vs-length-vs-size-and-what-will-happen-if-you-use-it-the-way-you-shouldnt/